stdlib/Dates/src/periods.jl

```diff
@@ -9,16 +9,44 @@ For a given period, return the value associated with that period.  For example,
 """
 value(x::Period) = x.value
 
+abbr(::Type{Nanosecond}) = "ns"
+abbr(::Type{Microsecond}) = "μs"
+abbr(::Type{Millisecond}) = "ms"
+abbr(::Type{Second}) = "s"
+abbr(::Type{Minute}) = "m"
+abbr(::Type{Hour}) = "h"
+abbr(::Type{Day}) = "d"
+abbr(::Type{Week}) = "w"
+abbr(::Type{Month}) = "mo"
+abbr(::Type{Quarter}) = "q"
+abbr(::Type{Year}) = "yr"
```

Comment:
I personally really do not like the look of using `yr`s instead of `years`, `q` instead of `quarters` or `m` instead of `meters`. The parsing for abbreviations seems okay, though, there are more abbreviations than those, so seems like it wants a sort of prefix comparison? e.g. allowing `y` for year and `wk` for week